src/read_command.rs

```diff
@@ -0,0 +1,88 @@
+use crate::webhook::Webhook;
+
+#[derive(PartialEq, Debug)]
+pub struct BotCommand {
+    pub command: String,
+    pub bot: String,
+}
+
+pub fn extract_commands(webhook: &Webhook, bots: &Vec<String>) -> Vec<BotCommand> {
+    // prevent bots from triggering commands potentially creating loops
+    if bots.contains(&webhook.author) {
+        println!("{} cannot trigger commands", webhook.author);
+        return vec![];
+    }
+
+    // only continue if comment was not deleted
+    if webhook.action.eq("deleted") {
+        println!("Exiting: comment deleted");
+        return vec![];
+    }
+
+    // ensure author of command has sufficient rights
+    let rights = &webhook.author_association;
+    if !(rights.eq("OWNER") || rights.eq("COLLABORATOR")) {
+        println!("permission denied: {}", rights);
+        return vec![];
+    }
+
+    parse_commands(&webhook.comment, bots)
+}
+
+fn parse_commands(text: &str, bots: &Vec<String>) -> Vec<BotCommand> {
+    let mut bot_commands: Vec<BotCommand> = Vec::new();
+    for bot in bots {
+        let atbot = &format!("@{bot} ");
+        text.lines()
+            .filter(|line| line.starts_with(atbot))
+            .flat_map(|s| -> Vec<String> {
+                s.to_string()
+                    .replace(atbot, "")
+                    .split(' ')
+                    .map(|s| s.to_string())
+                    .collect()
+            })
```

Comment:
You can keep this as it is, but I just wanted to mention that there is a cool function [`str::strip_prefix`](https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix) that could be useful here.  Also, it is most often not necessary to `collect` inside chained iterators:

```suggestion
        text.lines()
            .flat_map(|line| line.strip_prefix(atbot))
            .flat_map(|s| s.split(' '))
```

The first `flat_map` filters for lines starting with `atbot` and removes it at the same time.  The second `flat_map` replaces the `collect` and produces a single iterator over the commands.

Reply:
This is a lot nicer thank you!
Also yes, as this was supposed to be as general purpose as possible, i think you are right in suggesting to add arguments to commands. This is added now.